src/main/scala/org/renci/cam/Utilities.scala

```diff
@@ -27,6 +29,29 @@ object Utilities {
           .collect {
             case (key, Right(value)) => (key, value)
           }
-    } yield curies
+    } yield PrefixesMap(curies)
+
+  def getBiolinkPrefixesFromFile: ZIO[Any, Throwable, PrefixesMap] = for {
+    val prefixesStr = fromFile("prefixes.json").mkString
+    val prefixesJson = parse(prefixesStr)
+    val cursor = prefixesJson.hcursor
+    val contextValue = ZIO.fromEither(cursor.downField("@context").as[Map[String, Json]])
+    val curies =
+      contextValue
+        .map {
+          case (key, value) => (key, value.as[String])
+        }
+        .collect {
+          case (key, Right(value)) => (key, value)
+        }
+  } yield curies //getResourcesAsStream
+
+  def getPrefixes: ZIO[HttpClient, Throwable, PrefixesMap] = getBiolinkPrefixesFromURL.orElse(getBiolinkPrefixesFromFile)
+
+  def makePrefixesLayer: UIO[TaskLayer[Has[PrefixesMap]]] = getBiolinkPrefixesFromURL.map(ZLayer.fromManaged())
+
+  val biolinkPrefixes: ZIO[HttpClient, Nothing, TaskLayer[Has[PrefixesMap]]] = ZIO.service
+
+  case class PrefixesMap(prefixesMap: Map[String, String])
```

Comment:
@balhoff Does `case class` work, or should I implement a separate class?

Reply:
That works. Make all case classes final though: https://nrinaudo.github.io/scala-best-practices/tricky_behaviours/final_case_classes.html